### ISSUE TITLE
Windows Defender detects v1.5 as a Trojan

### ISSUE THREAD

**Issue by alexdelprete** [HerMajestyDrMona/Windows11DragAndDropToTaskbarFix#10]
v1.4 does not trigger Defender, v1.5 does. :(

![image](https://user-images.githubusercontent.com/7027842/139070907-48212451-1a27-4cf2-bafe-d9518b374624.png)

**Comment by HerMajestyDrMona** [comment]
@alexdelprete Hi. Did you experience similar problems again? I recently noticed that the bug randomly appears, for example when disconnecting the secondary screen, or when plugging the battery charger. I'll be releasing a fix for these issues too in next days, but just wanted to ask if you're still experiencing these issues?

**Comment by alexdelprete** [comment]
Hi. No but I noticed a small issue when there are two windows of the same app: my mail client, when I open a mail, it opens two windows and they stay in the same icon on the taskbar, when you drop something into the icon, it opens the two subicons, but it's very very hard to pick the second subicon. I was dropping a file in an email.

Regarding what you noticed: I don't have a second monitor and my laptop is always plugged in, so I didn't notice the issue.

**Comment by HerMajestyDrMona** [comment]
@alexdelprete Thanks for the reply. Do you have this problem with other apps that use multiple windows under 1 icon, for example multiple windows of Windows Explorer?

**Comment by alexdelprete** [comment]
Yes, same problem with WE. It's difficult to pick the subwindow. You have 0.5s to pick it, if you miss that time window, it gets difficult. I think you can easily reproduce it.

**Comment by HerMajestyDrMona** [comment]
@alexdelprete I can't reproduce it. It is supposed to first auto-open the first window, and then show the list of windows again. Did you modify the configuration (something else than Autostart or Hide Console Window?

**Comment by alexdelprete** [comment]
Nope, I used the default config file but didn't modify any value.

**Comment by HerMajestyDrMona** [comment]
@alexdelprete Ok, so just to confirm: You can never get the preview of windows under a single icon? It just opens the first window from the group and doesn't return to show others?

To fix your problem, please use:
```
AutoOpenFirstWindowInBestMethodEver=0
```

for now. And I'll try to reproduce it. But might be hard to do so as it works as expected for me :(

**Comment by alexdelprete** [comment]
> You can never get the preview of windows under a single icon? It just opens the first window from the group and doesn't return to show others?

No, the windows both open, but I have a hard time picking the second window, it's a timing issue.

> This way we will know if it's the timing problem or something else

Exactly, it's a timing issue. I'll try your suggested values and report back. :)

**Comment by HerMajestyDrMona** [comment]
> No, the windows both open, but I have a hard time picking the second window, it's a timing issue.

It should work the following way:
- Display all windows under the app icon.
- After X milliseconds open the first window.
- Again display all windows under the app icon.

I think it fails for you at step 3. Adding a longer `HowLongSleepAfterAutoOpenFirstWindowMilliseconds` might fix it. So let's see.

I'm finally adding the tray icon for this program, so should be easier to manage / restart / quit.

**Comment by alexdelprete** [comment]
> * After X milliseconds open the first window.

That was my problem, solved it with: `AutoOpenFirstWindowInBestMethodEver=0`

I don't want it to auto-open any window unless I drop the file in the window I choose. :)

Thanks a lot for the explanation, it put me in the right direction.

**Comment by HerMajestyDrMona** [comment]
@alexdelprete Could you please revert `AutoOpenFirstWindowInBestMethodEver=1` and try `HowLongSleepAfterAutoOpenFirstWindowMilliseconds=500` to see if after the first window auto-opens it starts displaying thumbnails of all windows under the icon again? Because from what I understood it wasn't working for you. It's just for the test purposes, you can disable the auto-open later :)

**Comment by alexdelprete** [comment]
It works also with `HowLongSleepAfterAutoOpenFirstWindowMilliseconds=100`. The problem was the confusion that the auto-open created. Actually, it's better with 100 because I see the thumbnails immediately.

I just don't want it to auto-open. Actually, I'd like to auto-open if there's only 1 item (you can't choose), but if >1, then don't auto-open.

**Comment by HerMajestyDrMona** [comment]
Oh yes. I wasn't a fan of auto-opening, but people on reddit wanted it.

Yes, I was planning to add a separate option to make it work only when 1 window is present. However, it will require some calculations that terrify me. Generally I'll need to get the width and the position of the window that displays thumbnails, then somehow determine if it's a greater size than 1 window (and make sure it's shown above the certain icon). I'm not sure if the size of this window is the same on all DPI scaling settings, etc. Anyway I was going to try it :)

**Comment by alexdelprete** [comment]
There's no function that tells you how many items are there? wow...:(

Anyway, I'm good like this, don't want you to get crazy for that functionality. Unless you like it too. :)

**Comment by HerMajestyDrMona** [comment]
> There's no function that tells you how many items are there? wow...:(

The program in 95% is "guessing" things. But at the current stage it seem to do it pretty well. I'm scared to start rewriting the code to "beauty" it a little.

And yes, I wanted that limited-auto-open option myself.

**Comment by alexdelprete** [comment]
> The program in 95% is "guessing" things. But at the current stage it seem to do it pretty well. I'm scared to start rewriting the code to "beauty" it a little.

Let me tell you that you're an experienced "guesser", because it's working great. :)

> And yes, I wanted that limited-auto-open option myself.

Great. I was thinking about the value of `AutoOpenFirstWindowInBestMethodEver`, 0 is disabled, 1 is enabled, and 2 could be limitedly-enabled. :)

**Comment by HerMajestyDrMona** [comment]
@alexdelprete Please check [version 1.7.0.0](https://github.com/HerMajestyDrMona/Windows11DragAndDropToTaskbarFix/blob/main/CHANGELOG.md#ver-1700----2021-10-31-1710-cet). There's a tray icon and the program by default won't auto-open if there are 2 or more windows under the same  icon:  `AutoOpenFirstWindowInBestMethodEver=2` (you don't need to keep this option in your config file, as it's default). Please also let me know if Windows Defender is crying again. For me it didn't :)

Also thank you so much for discussion and providing information. I very appreciate help in making the program better. Hopefully soon we won't have any issues to discuss about :)

**Comment by alexdelprete** [comment]
Thanks, it's working fine. Since I had the config file, I changed it to 2 anyways before running it. :)

Thank you for the discussion, I appreciate your kindness and open-mindness, not many devs listen to users suggestions and feedback.

Will let you know if some other ideas come up or I catch some glitches. :)

**Comment by HerMajestyDrMona** [comment]
No problem. I'm not a professional programmer, just in hobby I code something (I never went to the IT school or anything like this). So maybe this is the reason why I sound a bit differently.

The fix was mostly created for myself, because I wouldn't be able to suffer without this feature, and I didn't want to go back to Windows 10 either. I thought that it will be a simple thing (and it was), but once this software got so popular (over 200 stars now!) I decided to improve it a bit more :) I also didn't expect that there will be YouTube tutorial videos made by people about this program. These things are really motivating.

Thanks again for your feedback.

**Comment by alexdelprete** [comment]
Well, I recognized that, I'm in the IT industry since I was born, have done every role in IT, also Project Manager with teams of developers, and let me tell you that the most creative and skillful devs I found didn't study IT at all, they came from math, physics and even arts studies. I think their advantage is not being indoctrinated in UNIs by the usual stuff, so they were far more efficient in solving problems. And I can see that aspect in you...:)

I took a look at the source code, and it's not a "fix" as you call it, it's a nice piece of creative software. And to do that you surely had to know Windows APIs and stuff, so don't push down what you've accomplished. I, like many other users, am really grateful for what you did, and the support you give.

Keep up the great job, I just installed v1.8.0. :)
